syntax = 'proto3';
package grpc_demo;

option go_package = ".;main";

message CastMember {
  string character = 1;
  string firstName = 2;
  string lastName = 3;
} 

message Movie {
  string title = 1;
  string description = 2;
  int32 productionYear = 3;
  string genre = 4;
  int32 duration = 5;
  repeated CastMember castMembers = 6;
}

message MovieCatalogue {
  repeated Movie movies = 1;
}
